server/app.js

```diff
@@ -4,42 +4,43 @@ const favicon = require('serve-favicon');
 const logger = require('morgan');
 const cookieParser = require('cookie-parser');
 const bodyParser = require('body-parser');
-const mongo = require('mongodb');
+const MongoClient = require('mongodb').MongoClient;
 
 const api = require('./routes/api');
 
 const app = express();
 
-// uncomment after placing your favicon in /public
-//app.use(favicon(path.join(__dirname, 'public', 'favicon.ico')));
-app.use(logger('dev'));
-app.use(bodyParser.json());
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(cookieParser());
-app.use(express.static(path.join(__dirname, 'public')));
-
-app.use('/api', api);
-
-// catch 404 and forward to error handler
-app.use(function(req, res, next) {
-  let err = new Error('Not Found');
-  err.status = 404;
-  next(err);
+var mongoUtil = require('./helpers/mongoUtil.js');
+
+mongoUtil.connectToServer(function(err){
+
```

Comment:
The contents of this callback just set up the server and don't have to deal with actually starting up the server. If you look at `bin/www`, `http.createServer` is what's starting it up. So that is what should be wrapped. Otherwise, someone could technically start hitting the server right after it starts up and before it connects to MongoDB